edx_rbac/mixins.py

```diff
@@ -61,3 +65,79 @@ def check_permissions(self, request):
             self.permission_denied(
                 request,
                 message=('MISSING: {}'.format(', '.join(missing_permissions))))
+
+
+class PermissionRequiredForListingMixin(PermissionRequiredMixin):
+    """
+    Mixin for checking permissions against multiple objects.
+    """
+    list_lookup_field = None
+    staff_are_never_forbidden = True
+    allowed_roles = []
+    role_assignment_class = None
+
+    @property
+    def base_queryset(self):
+        raise NotImplementedError
+
+    @cached_property
```

Comment:
Out of curiosity (and for my own learning), how do you decide between when to use `@property` or `@cached_property`?

Reply:
I use `cached_property` when it's something expensive to compute that I don't need to re-compute later.  I use `property` when it's cheap to compute, or if its return value might change between invocations of it (and I care about the change).